### PR TITLE
BREAKING CHANGE: make version required

### PR DESCRIPTION
**Pull request by nchamo**
We are now making the entity's version a required field in both `Entity` and `Deployment`

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -26,13 +27,12 @@ export type Entity = {
...
 }
 
-export type ServerVersion = string
 export type ServerName = string
 export type ServerAddress = string
 
 export type ServerStatus = {
   name: ServerName
-  version: ServerVersion
+  version: EntityVersion
```
Comment (agusaldasoro) [inline src/types.ts:35]: This would be something like `default version`?

File: src/utils/EntityFactory.ts [hunk 2]
```diff
@@ -25,7 +47,10 @@ export async function buildEntityAndFile(
   }
 
   const entityFile = Buffer.from(JSON.stringify(entity))
-  const entityId: EntityId = await Hashing.calculateBufferHash(entityFile)
+  const entityId: EntityId =
+    version === EntityVersion.V3
```
Comment (agusaldasoro) [inline src/utils/EntityFactory.ts:51]: Shouldn't we ask for v4 instead? As we are sure that it's present in the new version

**Review by agusaldasoro — APPROVED** [review]

File: src/types.ts (hunk 1 above)
Comment (nchamo) [reply src/types.ts:35]: I don't understand the question. `EntityVersion` is an enum on this same file

File: src/utils/EntityFactory.ts (hunk 2 above)
Comment (nchamo) [reply src/utils/EntityFactory.ts:51]: I don't understand the comment. The idea is that if the version is v3, we will use the current hashing method. If the version is v4 or higher, we will use ipfs

File: src/types.ts (hunk 1 above)
Comment (agusaldasoro) [reply src/types.ts:35]: I wasn't sure what this field was used for, but now I get it!

**Pull request merged**